localstack-core/localstack/services/cloudformation/engine/v2/change_set_model_executor.py

```diff
@@ -89,10 +89,9 @@ def visit_node_depends_on(self, node_depends_on: NodeDependsOn) -> PreprocEntity
         if array_identifiers_delta.after:
             depends_on_resource_logical_ids.update(array_identifiers_delta.after)
         for depends_on_resource_logical_id in depends_on_resource_logical_ids:
-            node_resource = self._get_node_resource_for(
-                resource_name=depends_on_resource_logical_id, node_template=self._node_template
-            )
-            self.visit_node_resource(node_resource)
+            for node_resource in self._node_template.resources.resources:
```

Comment:
Why are we avoiding `_get_node_resource_for`? It looks like the user can define `depends_on` values that don't match a resource in the template

Reply:
After discussion with @MEPalma I replaced this removal of `_get_node_resource_for` with the original function, but using `self.visit` instead of `self.visit_node_resource` as `self.visit` implements the expected caching behaviour to prevent double-deployment of resources.